Assets/Scripts/Localization/LocalizationDownloader.cs

```diff
@@ -6,403 +6,366 @@
-            if (index + 1 >= githubApiResponse.Length - 1)
-            {
-                // Either the response was damaged or GitHub API returned an error.
-                throw new Exception("Error at parsing JSON - githubApiResponse too short.");
-            }
+                            // Let us make sure the file exists before we try to modify it.
+                            if (!File.Exists(path))
+                            {
+                                Debug.ULogErrorChannel("LocalizationDownloader", "Error while patching file " + file.Filename + " for the localization. The file doesn't exist, so I'll download it.");
+                                DownloadLocalization(new string[] { file.Filename }, hash);
+                                break;
+                            }
 
-            char currentChar = githubApiResponse[index];
+                            // Patch the file
+                            List<Patch> patches = Patcher.PatchFromText(file.Patch);
+                            object[] patch = Patcher.PatchApply(patches, File.ReadAllText(path));
 
-            // Hash of the commit.
-            string hash = string.Empty;
+                            // Let's see if the patch applied correctly.
+                            bool[] boolArray = (bool[])patch[1];
+                            if (boolArray.Length == 0 || !boolArray[0] || !boolArray[1])
+                            {
+                                Debug.ULogErrorChannel("LocalizationDownloader", "Error while patching file " + file.Filename + " for the localization. I'll just download it.");
+                                DownloadLocalization(new string[] { file.Filename }, hash);
+                                yield break;
+                            }
 
-            // Get the commit hash
-            do
-            {
-                hash += currentChar;
-                index++;
-                currentChar = githubApiResponse[index];
+                            File.WriteAllText(path, patch[0].ToString());
+                            break;
 
-                // Check if this is the end of the commit string.
-                if (index + 1 == githubApiResponse.Length - 1)
-                {
-                    // Either the response was damaged or GitHub API returned an error.
-                    throw new Exception("Error at parsing JSON - hash closing tag not found.");
+                        case "added":
+                            Debug.ULogChannel("LocalizationDownloader", "Adding the file " + file.Filename + " to the localization.");
+                            
+                            DownloadLocalization(new string[] { file.Filename }, hash);
+                            break;
+                        
+                        case "renamed":
+                            Debug.ULogErrorChannel("LocalizationDownloader", "Error, we tried to rename the file " + file.Filename + " but that isn't supported yet!");
```

Comment:
Have this option default to completely reinstalling a fresh copy of localization until this is properly implemented. otherwise this could cause interesting bugs.